### PR TITLE
CPR 188 reduce test mocking

### PR DESCRIPTION
**Pull request by markreesmoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt [hunk 1]
```diff
@@ -157,30 +151,4 @@ class CourtCaseEventsServiceTest {
     verify(personRecordService, never()).createNewPersonAndDefendant(person)
     verify(telemetryService).trackEvent(TelemetryEventType.HMCTS_PARTIAL_MATCH, mapOf("Surname" to "Jones"))
   }
-
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt:160]: https://github.com/ministryofjustice/hmpps-person-record/blob/87dbc4ead36aecd403d565e5170aec75e4324bc9/src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt#L55 covers this functionality

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt [hunk 2]
```diff
@@ -57,21 +51,6 @@ class CourtCaseEventsServiceTest {
...
-    // When
-    courtCaseEventsService.processPersonFromCourtCaseEvent(person)
-
-    // Then
-    verify(personRecordService, never()).createNewPersonAndDefendant(person)
-    verify(telemetryService).trackEvent(TelemetryEventType.MISSING_PNC, emptyMap())
-    verify(telemetryService, never()).trackEvent(INVALID_PNC, mapOf("PNC" to ""))
-    verifyNoMoreInteractions(telemetryService)
-  }
-
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt:74]: This is covered here:

https://github.com/ministryofjustice/hmpps-person-record/blob/9b6ebaba994608a1e47a9fe3dcb396d22d65d08c/src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt#L88

**Review by vb-moj — DISMISSED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt [hunk 3]
```diff
@@ -32,61 +30,17 @@ class CourtCaseEventsServiceTest {
...
-    // When
-    courtCaseEventsService.processPersonFromCourtCaseEvent(person)
-
-    // Then
-    verify(personRecordService, never()).createNewPersonAndDefendant(person)
-    verify(telemetryService).trackEvent(TelemetryEventType.MISSING_PNC, emptyMap())
-    verifyNoMoreInteractions(telemetryService)
-  }
-
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt:59]: Moved into integration test

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt [hunk 4]
```diff
@@ -157,30 +111,4 @@ class CourtCaseEventsServiceTest {
...
-    // When
-    courtCaseEventsService.processPersonFromCourtCaseEvent(person)
-
-    // Then
-    verify(personRecordService).createNewPersonAndDefendant(person)
-    verify(telemetryService).trackEvent(TelemetryEventType.HMCTS_RECORD_CREATED, mapOf("UUID" to uuid.toString(), "PNC" to pncNumber))
-    verify(offenderService).processAssociatedOffenders(personEntity, person)
-    verify(prisonerService).processAssociatedPrisoners(personEntity, person)
-  }
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt:185]: Covered here: https://github.com/ministryofjustice/hmpps-person-record/blob/9b6ebaba994608a1e47a9fe3dcb396d22d65d08c/src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt#L183

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt [hunk 5]
```diff
@@ -32,99 +27,17 @@ class CourtCaseEventsServiceTest {
...
-
-    // When
-    courtCaseEventsService.processPersonFromCourtCaseEvent(person)
-
-    // Then
-    verify(personRecordService, never()).createNewPersonAndDefendant(person)
-    verify(telemetryService).trackEvent(VALID_PNC, mapOf("PNC" to pncNumber))
-    verify(telemetryService).trackEvent(HMCTS_EXACT_MATCH, mapOf("PNC" to pncNumber, "CRN" to crn, "UUID" to personID.toString()))
-  }
-
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsServiceTest.kt:127]: Moved into integration test "should output correct telemetry for exact match"

**Pull request merged**